test/e2e/operator_test.go

```diff
@@ -495,12 +497,10 @@ func TestUpdateDefaultIngressController(t *testing.T) {
 	}
 
 	// Reset .spec.defaultCertificate to its original value.
-	if err := kclient.Get(context.TODO(), defaultName, ic); err != nil {
-		t.Fatalf("failed to get default ingresscontroller: %v", err)
-	}
-	ic.Spec.DefaultCertificate = originalSecret
-	if err := kclient.Update(context.TODO(), ic); err != nil {
-		t.Errorf("failed to reset default ingresscontroller: %v", err)
+	if err := updateIngressControllerSpecWithRetryOnConflict(t, defaultName, timeout, func(spec *operatorv1.IngressControllerSpec) {
+		spec.DefaultCertificate = originalSecret
+	}); err != nil {
+		t.Fatalf("failed to reset default ingresscontroller: %v", err)
```

Comment:
I don't see a problem with this change per se, but the preceding logic is questionable.  `TestUpdateDefaultIngressController` could set `spec.DefaultCertificate`, fail to observe the expected change in the deployment or fail to observe the expected change in the configmap, and terminate the test before reverting `spec.DefaultCertificate` back to the original value `originalSecret`.  Lines 480 and 496 in this file in this commit are instances where we should be using `Error` instead of `Fatal`.  (Alternatively, we could have a deferred function to reset `spec.DefaultCertificate` if the test doesn't reach line 500 to to a `Fatal` on line 480 or line 496.)  We can fix that in a follow-up though.

Reply:
I'd also like to explore helper functions that return a resource but internal to the implementation of the helper  make use of [`t.Cleanup`](https://pkg.go.dev/testing#T.Cleanup) in lieu of more defer functions. But that effort is also for another follow-up.